### DISCUSSION TITLE
Conditional mappings if LSP is running

### DISCUSSION THREAD

**Discussion by pbnj** [w0rp/ale#4224] [Q&A - Ask for help with problems] [answered]
I would like to set up some mappings only if an LSP server is running.

For example:

```viml
if g:ale_lsp_running == 1
  nnoremap gd <cmd>ALEGoToDefinition<cr>
  nnoremap K  <cmd>ALEHover<cr>

  " ... and other LSP-specific mappings for ALEGoToTypeDefinition, ALEGoToImplementation, ALECodeAction, ALEFindReferences, ...etc

endif
```

I have briefly looked through the docs, but didn't find anything relevant that would allow me to check if an LSP is running.

Appreciate any ideas/help.

**Comment by hsanson** [comment, marked as answer]
I have this in my config. Whenever a LSP is started the ALELSPStarted event is triggered so I have an autocommand that call the ALELSPMappings() to do some configurations.

Note this event is called everytime a LSP is started so make sure the method you call can be invoked several times without issues.


```
function ALELSPMappings()
  nnoremap <buffer> gk :ALEDocumentation<cr>
  nnoremap <buffer> gr :ALEFindReferences<cr>
  nnoremap <buffer> gd :ALEGoToDefinition<cr>
  nnoremap <buffer> gy :ALEGoToTypeDefinition<cr>
  nnoremap <buffer> gh :ALEHover<cr>
  setlocal omnifunc=ale#completion#OmniFunc
endfunction

augroup ALEMappings
  autocmd!
  autocmd User ALELSPStarted call ALELSPMappings()
augroup END
```

**Reply by pbnj** [reply]
Thank you. This works for my needs.
